pkg/plugin/plugin_agent.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: Cristian Mogildea
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+func StartContainer(cfg *Config) error {
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return err
+	}
+
+	authStr, err := GetRegistryAuth()
+	if err != nil {
+		return err
+	}
+
+	reader, err := cli.ImagePull(ctx, cfg.DockerImg, types.ImagePullOptions{RegistryAuth: authStr})
+	if err != nil {
+		return err
+	}
+	io.Copy(os.Stdout, reader)
+
+	containerCmd := GenerateCmd(cfg)
+
+	resp, err := cli.ContainerCreate(ctx,
```

Comment:
Func is already long. I would extract the `cli.ContainerCreate(...)` operation into a own func that accepts the plugin config as argument.

```go
resp, err := containerCreate(cfg)
```

Reply:
My next commit fixes this